public/app/features/alerting/unified/components/rule-editor/DashboardPicker.tsx

```diff
@@ -145,7 +133,7 @@ export const DashboardPicker = ({ dashboardUid, panelId, isOpen, onChange, onDis
   const PanelRow = ({ index, style }: { index: number; style: CSSProperties }) => {
     const panel = filteredPanels[index];
     const panelTitle = panel.title || '<No title>';
-    const isSelected = panel.id && selectedPanelId === panel.id?.toString();
+    const isSelected = !!panel.id && Number(selectedPanelId) === panel.id;
```

Comment:
nit: Prefer `Boolean()` over `!!`

Reply:
nit: do we still need to cast to a number here? Seems like `selectedPanelId` is already typed as `number | undefined`